llvm/lib/Transforms/IPO/FunctionSpecialization.cpp

```diff
@@ -468,16 +468,25 @@ Constant *InstCostVisitor::visitCastInst(CastInst &I) {
 Constant *InstCostVisitor::visitCmpInst(CmpInst &I) {
   assert(LastVisited != KnownConstants.end() && "Invalid iterator!");
 
+  Constant *Const = LastVisited->second;
```

Comment:
It's not your fault, but I do feel the 'Swap' variable name is perhaps a bit misleading here. It seems to me that really the variable is true if the specialised value (that will be replaced by a constant) is on the RHS. It's almost like the variable should just be called `ConstOnRHS` or something like that.

Reply:
Hey folks, apologies for the misleading variable. If I remember correctly there are more visit functions with the same logic. Can we change those too?